enhancements/validate-catalog-for-bad-bundles.md

```diff
@@ -0,0 +1,217 @@
+- Not a lot to offer in terms of flexibility as compared to [OpenPolicyAgent (OPA)](https://kubernetes.io/blog/2019/08/06/opa-gatekeeper-policy-and-governance-for-kubernetes/). With OPA, you can enforce custom policies for create, update and delete of Kubernetes objects without recompiling or reconfiguring the Kubernetes API server or even Kubernetes Admission Controllers. 
+- Require users supplying CUE validation template to be familiar with a new type of language.
+- Documentation is vague and doesn't have concrete implementation examples
+- The error messages are very basic and not customizable(AFAIK)
+
+
+## Part 2: Tool to list "bad" bundles in a catalog
+The basic approach will be to:
+- Extend [`opm`](https://github.com/operator-framework/operator-registry/blob/master/docs/design/opm-tooling.md) tooling to add two new commands:
+    - `opm index extract <catalog-image>` - pull and unpack catalog image
+    - `opm index validate <registry file location>` - validate all the bundle images in the registry file
+
+### Extract the contents of a catalog source image 
+Pull the catalog image specified by the <catalog image> positional argument to extract the database file and store it in a location specified by `--download-folder`
+```shell=
+opm index extract quay.io/operator-framework/upstream-community-operators:latest --download-folder ./index-manifest
+```
+
+- The first positional argument points to the catalog source image which has the running index image
+- `--download-folder` flag points to a directory location where the extracted file(s) gets stored
+
+#### Define a new API to unpack and store the mainfests
+- Define a struct that stores the command line arguments from the `opm index extract` command. 
+```go=
+type ExtractFromIndexRequest struct {
+	Index         string
+	DownloadPath  string
+	ContainerTool containertools.ContainerTool
+}
+```
+- Add `ExtractFromIndex` API to [registry/interfaces.go](https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/registry/interfaces.go) by defining a conterfieter
+```go=
+type IndexExtracter interface {
+	ExtractFromIndex(ExtractFromIndexRequest) error
+}
+```
+- This API reads the `ExtractFromIndexRequest` and unpacks the index image and extracts the database file (similar to [this](https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/indexer/indexer.go#L453))
+
+### Validate each bundle image from the registry file
+```shell=
+$ opm index validate <registry file location>
+```
+- The command reads the registry file to get the list of bundle images and unpacks each bundle image to it's own directory (ToDo: Should I consider only distinct bundle images or everything?)
+- Define an API which performs bundle validation (possibly reusing [this code from line 63](https://github.com/operator-framework/operator-registry/blob/master/cmd/opm/alpha/bundle/validate.go#L63)) 
```

Comment:
I would expect this command to be identical to `opm alpha bundle validate`, but with the ability to specify a directory with a collection of unpacked bundles in it rather than a single bundle.